src/Inventory1.java

```diff
@@ -0,0 +1,198 @@
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Implementation of Inventory on Array and Set.
+ *
+ * @convention <pre>
+ * |$this.slots| > 0 and
+ * for all 0 <= i < |$this.slots|:
+ *  [$this.slots[i] is defined and is not null] and
+ *  [if $this.slots[i] is not an empty Item, it has all the entries in this.reqs as tags]
+ * </pre>
+ *
+ * @correspondence <pre>
+ *  this = [the Items in $this.slots]
+ *</pre>
+ *
+ * @author David Stuckey
+ */
+public class Inventory1 extends InventorySecondary {
+
+    /** The primary representation variable: the slots of this. */
+    private Item[] slots;
+
+    /** The requirements for an Item to be added to this. */
+    private Set<String> reqs;
+
+    /**
+     * Creates initial representation.
+     *
+     * @param size
+     *            - the number of slots this will have
+     */
+    private void createNewRep(int size) {
+        this.slots = new Item[size];
+        this.reqs = new HashSet<String>();
+
+        for (int i = 0; i < size; i++) {
+            this.slots[i] = new BasicItem();
+        }
+    }
+
+    /**
+     * Creates a new Inventory with a single slot.
+     */
+    public Inventory1() {
+        this.createNewRep(1);
+    }
+
+    /**
+     * Create a new Inventory with one or more slots.
+     *
+     * @param size
+     *            the number of slots this will have
+     *
+     * @requires size > 0
+     */
+    public Inventory1(int size) {
+        assert size > 0 : "Violation of size > 0";
+
+        this.createNewRep(size);
+    }
+
+    //CHECKSTYLE: ALLOW THIS METHOD TO BE OVERRIDDEN
+    @Override
+    public void addItem(int slot, Item item) {
+        assert 0 <= slot : "Violation of 0 <= slot";
+        assert slot < this.slots.length : "Violation of slot < |this|";
+        assert item != null : "Violation of item is not null";
+        assert this.slots[slot].isEmpty() || this.slots[slot].equals(
+                item) : "Violation of slot is empty or has an Item of the same name.";
+        assert this.isAllowed(item) : "Violation of isAllowed(item)";
```

Comment:
Is this a violation of the kernel purity rule, since the other kernel method is simply used in an assertion?

Reply:
I think this is okay. Asserts are for development only. The compiler is going to remove them anyway.